app/models/piece.rb

```diff
@@ -7,11 +7,39 @@ def vectors(vectors = [])
     @vectors ||= vectors
   end
 
-  def move(x, y)
-    if self.move_capable?(position)
-      @x = x
-      @y = y
+  def move(params)
+    xpos = params[:x].to_i
+    ypos = params[:y].to_i
+    channel = "public-conversation"
+    if game.piece_at?(xpos, ypos)
+      captured_piece = game.piece_at(xpos, ypos)
+      captured_piece_fig = captured_piece.figure
+      captured_piece.update_attributes(x: 8, y:8, state: 'off')
+      Pusher.trigger(channel, 'move_event', {
+        piece: captured_piece,
+        fig: captured_piece_fig
+      })
     end
+    if type == 'King' && (self.x - xpos).abs == 2
+      if (self.x - xpos) == -2
+        castle_rook('king').update_attributes(x: 5, state: 'moved')
+        castle_rook = game.pieces.find_by(type: 'Rook', x: 5, y: self.y)
+        castle_rook_fig = castle_rook.figure
+        Pusher.trigger(channel, 'move_event', {
+          piece: castle_rook,
+          fig: castle_rook_fig
+        })
+      elsif (self.x - xpos) == 2
+        castle_rook('queen').update_attributes(x: 3, state: 'moved')
+        castle_rook = game.pieces.find_by(type: 'Rook', x: 3, y: self.y)
+        castle_rook_fig = castle_rook.figure
+        Pusher.trigger(channel, 'move_event', {
+          piece: castle_rook,
```

Comment:
Use 2 spaces for indentation in a hash, relative to the first position after the preceding left parenthesis.